File: langs/python/src/vmx/localization/localizer.py
"""ILocalizer Protocol. See spec/17-localization.md."""

from __future__ import annotations

from collections.abc import Iterable
from typing import Protocol, runtime_checkable


@runtime_checkable
class ILocalizer(Protocol):
    """Localization hook contract."""

    def localize(self, key: str, args: Iterable[object] | None = None) -> str:
        """Return the localized string for ``key``, optionally formatted with ``args``."""
        ...

File: langs/swift/Sources/VMx/Localization/Localizer.swift
/// Localization hook (spec/17-localization.md). A pure, stateless string lookup
/// contract — implementations may return localized text, the key, or "". Culture
/// changes are handled by swapping the injected Localizer (DI), not via a stream.
public protocol Localizer {
    /// Returns the localized string for `key`, optionally formatted with `args`.
    func localize(_ key: String, _ args: [Any]?) -> String
}

public extension Localizer {
    /// Convenience overload for the common no-args call.
    func localize(_ key: String) -> String { localize(key, nil) }
}
